### ISSUE TITLE
[java] UnnecessaryImport false positive when used in lambda

### ISSUE THREAD

**Issue by danjee** [pmd/pmd#4853]
**7.0.0-rc4** 

**Rule**: [UnnecessaryImport](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#unnecessaryimport)

**Code Sample demonstrating the issue:**

```java
.collect(groupingBy(
                                ext -> Product.valueOf(ext.getName()),
                                mapping(
                                        ext -> ext,
                                          //create a map with property and value
                                          Collectors.toMap(
                                                 ext -> ext.getProperty(),
                                                  Extension::getValue
                                        ))
                        )
                )
```

**No PMD violation will happen*

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** * Maven*

Problem is that not even 

`@SuppressWarnings({ "PMD.UnnecessaryImport"})`

helps.

**Comment by adangel** [comment]
Duplicate of #4816
